lib/propagation/aws.js

```diff
@@ -27,15 +27,19 @@ function marshalTraceContextv1(context) {
 exports.unmarshalTraceContext = unmarshalTraceContext;
 
 function unmarshalTraceContext(header) {
-  let traceId, parentSpanId;
+  let traceId, parentSpanId, customContext;
 
   const split = header.split(";");
   for (const s of split) {
     const [name, value] = s.split("=");
     if (name === "Root") {
       traceId = value;
-    } else if (name === "Parent") {
+    } else if (name === "Self") {
       parentSpanId = value;
+    } else {
+      customContext = Object.assign({}, customContext, {
+        [name]: value,
+      });
```

Comment:
maybe `customContext = {}` above, then:

```suggestion
    } else {
      customContext[name] = value
```

here?